src/coreclr/vm/jitinterface.cpp

```diff
@@ -14572,10 +14573,13 @@ void EECodeInfo::GetOffsetsFromUnwindInfo(ULONG* pRSPOffset, ULONG* pRBPOffset)
     }
 
     UNWIND_INFO * pInfo = GetUnwindInfoHelper(unwindInfo);
+
+    // Chained unwind info use violates assumptions of the security stackwalk cache.
+    // To avoid returning incorrect cache entries, invalidate the cache.
+    // Since chained unwind info is used only for cold main body code, this should not happen frequently.
     if (pInfo->Flags & UNW_FLAG_CHAININFO)
     {
-        _ASSERTE(!"GetRbpOffset() - chained unwind info used, violating assumptions of the security stackwalk cache");
-        DebugBreak();
+        StackwalkCache::Invalidate(pLoaderAllocator);
```

Comment:
This method is called before the entry is inserted into the cache, so there is nothing to invalidate at this point. 

Maybe a better approach is to ensure that the entry won't get inserted into the stackwalk cache when we hit this case. Ie suppress `m_crawl.stackWalkCache.Insert(&cacheEntry);` in stackwalk.cpp for this case.

Reply:
Thanks for catching that -- I'll push a revision with this approach.